File: internal/domain/evidence.go
package domain

import (
	"fmt"
	"strings"
)

// EvidenceReference links lore to supporting material.
type EvidenceReference struct {
	Type  EvidenceType
	Value string
}

// NewEvidenceReference validates and returns an evidence reference.
func NewEvidenceReference(evidenceType EvidenceType, value string) (EvidenceReference, error) {
	trimmed := strings.TrimSpace(value)
	if trimmed == "" {
		return EvidenceReference{}, validationError("evidence value must be non-empty")
	}
	if len(trimmed) > MaxEvidenceValueLength {
		return EvidenceReference{}, validationError(
			fmt.Sprintf("evidence value must be at most %d characters", MaxEvidenceValueLength),
		)
	}
	return EvidenceReference{Type: evidenceType, Value: trimmed}, nil
}

File: src/memlore/domain/models/evidence.py
from __future__ import annotations

from dataclasses import dataclass

from memlore.domain.exceptions import ValidationError
from memlore.domain.models.enums import EvidenceType

MAX_EVIDENCE_VALUE_LENGTH = 2048


@dataclass(frozen=True, slots=True)
class EvidenceReference:
    type: EvidenceType
    value: str

    def __post_init__(self) -> None:
        value = self.value.strip()
        if not value:
            raise ValidationError("evidence value must be non-empty")
        if len(value) > MAX_EVIDENCE_VALUE_LENGTH:
            raise ValidationError(
                f"evidence value must be at most {MAX_EVIDENCE_VALUE_LENGTH} characters"
            )
        object.__setattr__(self, "value", value)
